<launch>

<arg name="use_camera_info" default="false" doc="Indicates that the camera_info topic should be subscribed to to get the default input_frame_id. Otherwise the frame from the image message will be used." />
<arg name="debug_view" default="true" doc="Specify whether the node displays a window to show edge image" />

<!-- Crop Image Node -->
<node pkg="opt_flow" type="img_crop.py" name="image_crop" output="screen">
        <remap from="UBC_Image"  to="/camera1/image_raw" />
	    <param name="ymin" value="325" />
        <param name="ymax" value="425" />
        <param name="xmin" value="410" />
        <param name="xmax" value="620" />
        <rosparam>
     </rosparam>
</node>  


<!-- fback_flow.cpp -->
<node name="fback_flow" pkg="opencv_apps" type="fback_flow" >
    <remap from="image" to="/Crop_Image" />
    <param name="use_camera_info" value="$(arg use_camera_info)" />
    <param name="debug_view" value="$(arg debug_view)" />
</node>


<!-- Soil Estimation Node -->
--<node pkg="opt_flow" type="soil_est.py" name="soil_estimation" output="screen">
        <remap from="Crop_Flow"  to="/fback_flow/flows" />
	<rosparam>
     </rosparam>
</node> 
</launch>
